### PR TITLE
Mythic+ run history in tooltips

### PR DESCRIPTION
**Pull request by ntowle**
- Track and show mythic+ run history as relevant for the Great Vault in the SavedInstances tooltip
- Highlights runs which are considered for the Great Vault, and show associated item level
- Consider the "Mythic Keystone" and "Mythic Key Best" rows part of the same category

Examples:
image
image

### CODE REVIEW

**Comment by LiangYuxuan** [comment]
I would like to store entire weekly run history but showing top 10. Also for `.name` and `.rewardLevel`, add to old table rather than creating one, so we can keep `.mapChallengeModeID`.

**Comment by ntowle** [comment]
I could not fully re-test because the reset happened. Will hopefully get around to that in the next 24h.

One thing I noticed when testing the reset was the tooltip still shows a reward is waiting directly after claiming from the vault until a /reload or the vault is looked at again, even though C_WeeklyRewards.HasAvailableRewards() reports false. Maybe there's some event to listen to when claiming the weekly reward to trigger a re-check.

**Comment by LiangYuxuan** [comment]
I can confirm the run history part works well. Since there're not reset yet on CN realm so weekly reward part is not able to test.

**Comment by LiangYuxuan** [comment]
OKay, I've searched some blz code on PTR and found they use both `WEEKLY_REWARDS_SHOW` and `WEEKLY_REWARDS_UPDATE` to refresh their new map overlay.

Also, when they use their new global string `WEEKLY_REWARDS_RETURN_TO_CLAIM`, they checks on `C_WeeklyRewards.HasAvailableRewards()`, which means it's reward claimed but not selected. And `C_WeeklyRewards.CanClaimRewards()` means it's reward ready to claim.

https://github.com/tomrus88/BlizzardInterfaceCode/blob/7f62c8ace95343e09ca1e42d2d081b6eab950892/Interface/AddOns/Blizzard_WeeklyRewards/Blizzard_WeeklyRewards.lua#L176-L185

```lua
WEEKLY_REWARDS_RETURN_TO_CLAIM = "Return to the Great Vault to Claim your Reward";
```

So I think we should use

```lua
t.MythicKeyBest.rewardWaiting = C_WeeklyRewards_HasAvailableRewards() or C_WeeklyRewards_CanClaimRewards()
```

**Review by LiangYuxuan — APPROVED** [review]

**Review by LiangYuxuan — CHANGES_REQUESTED** [review]
Take back my approve for still having issue on after claiming weekly reward. I've done a fix for this, by delaying fetching due to no more event is in the claiming process but `WEEKLY_REWARDS_UPDATE`, tested on my alts and it works good. Also I try to make the threshold not hard-coded. This two commit are in my repo and you're free to use it :)

https://github.com/LiangYuxuan/SavedInstances/commit/95a519d0edd41b2628513e3797a15b8337dd41e0

https://github.com/LiangYuxuan/SavedInstances/commit/b274c9fcf9f7ae9ec5acf4f878f5c9a22ce27cbb

**Review by LiangYuxuan — APPROVED** [review]

File: SavedInstances/Modules/MythicPlus.lua [hunk 1]
```diff
@@ -90,21 +93,27 @@ do
   end
 
   local function runCompare(left, right)
-    return left.level > right.level
+    if left.level == right.level then
+      return left.mapChallengeModeID < right.mapChallengeModeID;
```
Comment (Stanzilla) [inline SavedInstances/Modules/MythicPlus.lua:97]: remove ;

File: SavedInstances/Modules/MythicPlus.lua [hunk 2]
```diff
@@ -90,21 +93,27 @@ do
   end
 
   local function runCompare(left, right)
-    return left.level > right.level
+    if left.level == right.level then
+      return left.mapChallengeModeID < right.mapChallengeModeID;
+    else
+      return left.level > right.level;
```
Comment (Stanzilla) [inline SavedInstances/Modules/MythicPlus.lua:99]: remove ;

**Review by Stanzilla — CHANGES_REQUESTED** [review]
just some small nits, otherwise lgtm!

**Review by Stanzilla — CHANGES_REQUESTED** [review]

**Review by Stanzilla — APPROVED** [review]

**Comment by Stanzilla** [comment]
Awesome, thank you!

**Pull request merged**